src/test/java/io/cdap/plugin/db/batch/action/vertice/export/VerticaBulkExportActionConfigTest.java

```diff
@@ -0,0 +1,152 @@
+    VerticaBulkExportAction.VerticaExportConfig config = getValidConfig();
+    config.path = null;
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(failureCollector);
+    assertValidationFailed(failureCollector,
+                           Collections.singletonList(VerticaBulkExportAction.VerticaExportConfig.PATH), 1);
+  }
+
+  @Test
+  public void testValidatePathEmpty() {
+    VerticaBulkExportAction.VerticaExportConfig config = getValidConfig();
+    config.path = "";
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(failureCollector);
+    assertValidationFailed(failureCollector,
+                           Collections.singletonList(VerticaBulkExportAction.VerticaExportConfig.PATH), 1);
+  }
+
+  @Test
+  public void testValidateUserNullAndPasswordNotNull() {
+    VerticaBulkExportAction.VerticaExportConfig config = getValidConfig();
+    config.user = null;
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(failureCollector);
+    assertValidationFailed(failureCollector,
+                           Collections.singletonList(VerticaBulkExportAction.VerticaExportConfig.USER), 1);
+  }
+
+  @Test
+  public void testValidateUserNotNullAndPasswordNull() {
+    VerticaBulkExportAction.VerticaExportConfig config = getValidConfig();
+    config.password = null;
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(failureCollector);
+    assertValidationFailed(failureCollector,
+                           Collections.singletonList(VerticaBulkExportAction.VerticaExportConfig.PASSWORD), 1);
+  }
+
+  @Test
+  public void testValidateConfig() {
+    VerticaBulkExportAction.VerticaExportConfig config = getValidConfig();
+    config.connectionString = null;
+    config.path = null;
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(failureCollector);
+    assertValidationFailed(
+      failureCollector, Arrays.asList(VerticaBulkExportAction.VerticaExportConfig.PATH,
+                                      VerticaBulkExportAction.VerticaExportConfig.CONNECTION_STRING), 2);
+  }
+
+  private static VerticaBulkExportAction.VerticaExportConfig getValidConfig() {
+    return new VerticaBulkExportAction.VerticaExportConfig(
+      "jdbc:vertica://localhost:5433/test",
+      "dbadmin",
+      "testpassword",
+      "select * from testTable",
+      ",",
+      "/tmp/vertica/vertice_test.csv");
+  }
+
+  private static void assertValidationFailed(MockFailureCollector failureCollector, List<String> paramName,
+                                             int failuresCount) {
+    List<ValidationFailure> failureList = failureCollector.getValidationFailures();
+    Assert.assertEquals(failuresCount, failureList.size());
+    for (int i = 0; i < failureList.size(); i++) {
```

Comment:
why don't you use foreach? Why do you need to check that failures were collected in same order as params in paramName list?